doc/adr/0004-use-yaml-file-for-bakery-database.md

```diff
@@ -0,0 +1,85 @@
+# 4. Use yaml file for Bakery database
+
+Date: 2021-03-29
+
+## Status
+
+Draft
+
+## Context
+
+We need to keep track of what bakeries exist as part of the system
+
+## Decision
+
+We will create a repository called `pangeo-forge/bakery-database`.
+Inside it, we will have a file called `bakeries.yaml`.
+
+### Example `bakeries.yaml`
+
+That file will look like this
+
+```yaml
+---
+aws_us_west2:
```

Comment:
Is this the name of the bakery?

I envision that `aws_us_west2` would be a pretty clash-able name, maybe we need to say that bakery names have to be a bit more specific/structured

Reply:
Pangeo Forge owns a bakery in Azure in `West Central US`:

`ge.pangeo_for.bakery.azure.west_central_us`

Reply:
Why you were writing this, I proposed `ldeo_aws_us_west2`. I appreciate your consideration of how this might scale. 

Just a minor point: Pangeo Forge is not a company, institution, etc. Each bakery will ultimately have to be owned by an institution or individual with a billing account.

Reply:
@rabernat Sure, it was just an out of the blue example. This structure you don't even require a website for your org, a github org could even have one.

Reply:
Could we handle that with extra metadata in this yaml?
```yaml
ldeo_aws_us_west2:
  operator_institution: Lamont Doherty Earth Observatory of Columbia University
  operator_website: "https://www.ldeo.columbia.edu/"
```
?

Reply:
I guess? We just need a defined schema for the name. Do we know how we're going about validating the database entries? I know we don't expect _many_ bakeries at the start, but an easy way to determine if your bakery name is unique enough would be good.

The idea behind the Java style is less about linking it back to an org and more about separating out similarly named packages.

Reply:
I just updated my example name to `edu.columbia.ldeo.bakery.aws.us-west2` and added a description about the syntax. I can see this verbose java syntax creeping in elsewhere.

I'm imagining the entire database will be in a single file, so conflicts should be obvious. We will need a parser / validator eventually.